dropwizard-unix-socket/src/test/java/io/dropwizard/unixsocket/UnixSocketConnectorFactoryTest.java

```diff
@@ -0,0 +1,124 @@
+import org.junit.jupiter.api.condition.OS;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.net.StandardProtocolFamily;
+import java.net.UnixDomainSocketAddress;
+import java.nio.channels.Channels;
+import java.nio.channels.SocketChannel;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+@DisabledOnOs(OS.WINDOWS) // Windows has Unix Socket support but not on all versions.
+@ExtendWith(DropwizardExtensionsSupport.class)
+class UnixSocketConnectorFactoryTest {
+    private static final String httpRequest = "GET /app/hello HTTP/1.1\r\n" +
+        "Host: dropwizard-unixsock\r\n" +
+        "\r\n";
+
+    public DropwizardAppExtension<Configuration> ext = new DropwizardAppExtension<>(HelloWorldApplication.class,
+        ResourceHelpers.resourceFilePath("yaml/usock-server.yml"));
+
+    private UnixDomainSocketAddress socketAddress;
+
+    @BeforeEach
+    public void setUp() {
+        final File socket = new File("/tmp/dropwizard.sock");
+        socket.deleteOnExit();
+
+        socketAddress = UnixDomainSocketAddress.of(socket.getPath());
+    }
+
+    @Test
+    public void testClient() throws Exception {
+        try (SocketChannel channel = SocketChannel.open(StandardProtocolFamily.UNIX)) {
+            channel.connect(socketAddress);
+            OutputStream os = Channels.newOutputStream(channel);
+            OutputStreamWriter osw = new OutputStreamWriter(os, UTF_8);
+            PrintWriter writer = new PrintWriter(osw);
+            InputStream is = Channels.newInputStream(channel);
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(is, UTF_8))) {
+                writer.print(httpRequest);
+                writer.flush();
+
+                verifyHelloWorldReceived(reader);
+            }
+        }
+    }
+
+    @Test
+    public void testManyCalls() throws Exception {
+        try (SocketChannel channel = SocketChannel.open(StandardProtocolFamily.UNIX)) {
+            channel.connect(socketAddress);
+            OutputStream os = Channels.newOutputStream(channel);
+            OutputStreamWriter osw = new OutputStreamWriter(os, UTF_8);
+            PrintWriter writer = new PrintWriter(osw);
+            InputStream is = Channels.newInputStream(channel);
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(is, UTF_8))) {
+                for (int i = 0; i < 1000; i++) {
+                    writer.print(httpRequest);
+                    writer.flush();
+
+                    verifyHelloWorldReceived(reader);
+                }
+            }
+        }
+    }
+
```

Comment:
The tests look reasonable but also rather complicated. I'm personally missing a client call. The Jetty client supports UNIX domain sockets, so we could add a test like this one:

```suggestion
    @Test
    void testHttpClientCall() throws Exception {
        ClientConnector clientConnector = ClientConnector.forUnixDomain(Paths.get("/tmp", "dropwizard.sock"));
        HttpClient httpClient = new HttpClient(new HttpClientTransportOverHTTP(clientConnector));
        httpClient.start();
        ContentResponse contentResponse = httpClient.GET("http://localhost:0/app/hello");
        assertThat(contentResponse).isNotNull().satisfies(response -> {
            assertThat(response.getContentAsString()).isEqualTo("{\"hello\": \"World\"}");
        });
        httpClient.stop();
    }

```

Reply:
ah, I couldn't find jersey client support earlier. I prefer jersey client test over the current one tbh. I'll change the existing ones instead.